cmd/api/api.go

```diff
@@ -106,14 +110,30 @@ type Service struct {
 
 // NewService creates a new API service.
 func NewService(cfg *config.ServerConfig) (*Service, error) {
-	logger := common.RootLogger().WithModule(moduleName)
+	ctx := context.Background()
+	logger := cmdCommon.RootLogger().WithModule(moduleName)
 
 	// Initialize target storage.
-	backing, err := common.NewClient(cfg.Storage, logger)
+	backing, err := cmdCommon.NewClient(cfg.Storage, logger)
 	if err != nil {
 		return nil, err
 	}
-	client, err := storage.NewStorageClient(cfg.ChainName, backing, logger)
+	runtimeClients := make(map[common.Runtime]storage.RuntimeClient)
+	if cfg.Node != nil {
+		apiRuntimes := []common.Runtime{common.RuntimeEmerald, common.RuntimeSapphire}
+		for _, runtime := range apiRuntimes {
+			client, err := source.NewRuntimeClient(ctx, cfg.Node, runtime)
+			if err != nil {
+				return nil, err
+			}
+			runtimeClients[runtime] = storage.RuntimeClient{
```

Comment:
Open to debate, but: I think it's needlessly layered to introduce a wrapper around RuntimeApiLite just to store the NativeTokenSymbol. Because that's all that the new RuntimeClient does, right? Encapsulation is nice, but lots of layers can also make it harder to read and navigate code.

An alternative approach, which I'd like to suggest here, is what analyzers already do (and mimicking their structure is a win on its own): Pass `cfg.Node` into NewStorageClient(), and then StorageClient can [implement](https://github.com/oasisprotocol/nexus/blob/b789dca343c4e81ce336b4354242fbe1ac3520dc/analyzer/runtime/runtime.go#L68-L71) `nativeTokenSymbol(runtime)` itself, basically reusing your one-liner from here.

Reply:
Valid points, updated! Note that the set of `nativeTokenSymbols` is decoupled from the set of `runtimeClients` but the former should be a superset of the latter so it should not be an issue.